doc/glossary.rst

```diff
@@ -0,0 +1,973 @@
+        class from a constructed instance. For example, in the following,
+        ``cls`` is an estimator class, while ``est1`` and ``est2`` are
+        instances::
+
+            cls = RandomForestClassifier
+            est1 = cls()
+            est2 = RandomForestClassifier()
+
+    examples
+        We try to give examples of basic usage for most functions and
+        classes in the API:
+
+        * as doctests in their docstrings (i.e. within the ``sklearn/`` library
+          code itself).
+        * as examples in the :ref:`example gallery <general_examples>`
+          rendered from scripts in the ``examples/`` directory, exemplifying
+          key features or parameters of the estimator/function.
+        * sometimes in the :ref:`User Guide <user_guide>` (built from ``doc/``)
+          alongside a technical description of the estimator.
+
+    evaluation metric
+        TODO
+
+    estimator tags
+        A proposed feature (e.g. :issue:`8022`) by which the capabilities of an
+        estimator are described through a set of semantic tags.  This would
+        enable some runtime behaviors based on estimator inspection, but it
+        also allows each estimator to be tested for appropriate invariances
+        while being excepted from other :term:`common tests`.
+
+        Some aspects of estimator tags are currently determined through
+        the :term:`duck typing` of methods like ``predict_proba`` and through
+        some special attributes on estimator objects:
+
+        ``_estimator_type``
+            This string-valued attribute identifies an estimator as being a
+            classifier, regressor, etc. It is set by mixins such as
+            :class:`base.ClassifierMixin`, but needs to be more explicitly
+            adopted on a :term:`meta-estimator`.  Its value should usually be
+            checked by way of a helper such as :func:`base.is_classifier`.
+
+        ``_pairwise``
+            This boolean attribute indicates whether the data (``X``) passed to
+            :func:`fit` and similar methods consists of pairwise measures over
+            samples rather than a feature representation for each sample.  It
+            is usually ``True`` where an estimator has a ``metric`` or
+            ``affinity`` or ``kernel`` parameter with value 'precomputed'.
+
+    feature
+    features
+        A feature is a function mapping a sampled object to a numeric or
+        categorical quantity.  "Feature" is also commonly used to refer to
+        these quantities, being the individual elements of a vector
+        representing a sample. In a data matrix, features are represented as
+        columns: each column contains the result of applying a feature
+        function to a set of samples.
+
+        Elsewhere features are known as attributes, predictors, regressors, or
+        independent variables.
+
+        Features in scikit-learn are generally expected to be numeric and
```

Comment:
Maybe "Nearly all estimators in scikit-learn assume..."? Also I would explicitly mention non-missing.